### PR TITLE
Adding a large number of rows won't break the table

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

The `alter` method used for inserting rows uses DataMap's `spliceData` method. As described [in the issue](https://github.com/handsontable/handsontable/issues/7840#issuecomment-825070300) method's body is transpiled by Babel. The `apply` method used after that has its limitations.

The easiest and probably the purest way of changing from transplied array's `splice` method is using another method. However, we need a method which **modify the original array**. Another solution will need some extra code for updating the reference. That's why `filterData` already execute `this.dataSource.length = 0` and update the array. 

https://github.com/handsontable/handsontable/blob/4c6d81235c51ca35251cf87f0f852a923010df90/handsontable/src/dataMap/dataMap.js#L624-L635

Changes has been already implemented by using `Array`'s `push` method in loop. Please keep in mind that alternative solution such as using spread operator for filling data's array is impossible for the same reasons as using the `splice` method. It's worth to mention also that [there is some issue](https://github.com/handsontable/handsontable/issues/9772) which shows another cases where source of a problem may be the same. This PR doesn't expand the scope and problems haven't been fixed.

I measured the performance for past and present solution and it seems that there is no significant difference in performance (the measured deviation was equal to `0,7%`). It's worth to mention also that there is no such problem with analogous `insert_col` option - it doesn't use `spliceData` method internally.

### How has this been tested?

Manually tested example from the issue. Measured the performance using loop and `console.time`.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7840 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/dataMap/dataMap.js [hunk 1]
```diff
@@ -596,14 +596,21 @@ class DataMap {
...
     if (continueSplicing !== false) {
-      this.dataSource.splice(index, amount, ...elements);
+      const newData = [...this.dataSource.slice(0, index), ...elements, ...this.dataSource.slice(index)];
+
+      // We try not to change the reference.
+      this.dataSource.length = 0;
+
+      // Pushing to array instead of using `splice`, because Babel changes the code to one that uses the `apply` method.
+      // The used method was cause of the problem described within #7840.
+      newData.forEach(row => this.dataSource.push(row));
```
Comment (jansiegel) [inline handsontable/src/dataMap/dataMap.js:613]: @wszymanski Have you checked if it does affect the performance in any way?

Comment (wszymanski) [reply handsontable/src/dataMap/dataMap.js:613]: As I wrote in PR description:

> I measured the performance for past and present solution and it seems that there is no significant difference in performance (the measured deviation was equal to 0,7%). It's worth to mention also that there is no such problem with analogous insert_col option - it doesn't use spliceData method internally.

🙂

Comment (jansiegel) [reply handsontable/src/dataMap/dataMap.js:613]: Oh, my bad, I missed that part 🙇

**Review by jansiegel — APPROVED** [review]

**Comment by aninde** [comment]
This fix works wonderfully on Chrome 104, FF 104.0.2, and Edge, but not on Safari. It is not related to the version, because on the older Safari 12.1 / Mojave Safari you can see it through the BrowserStack on demo https://jsfiddle.net/jkq38xdL/.
e2e doesn't pass there, due to the error in the browser console. I managed to cause an error locally after adding 65535 rows.

Screenshot 2022-09-14 at 16 10 11

Tested on Safari 15.6 (17613.3.9.1.5) / macOS 12.5, Safari 12.1 / Mojave with 35a0b036e9f6bdd1552d8d7fd69cf7cc4e35720b from freeze branch.

**Comment by wszymanski** [comment]
Good job, @aninde. I confirm the problem. I'm testing already implemented solution. I'll make another PR fixing the problem.

**Pull request merged**